### ISSUE TITLE
Detect version of former group ID on classpath

### ISSUE THREAD

**Issue by ennru** [akka/akka-persistence-jdbc#285]
### Short description

Detect if this plugin of the old 3.5.x series is on the classpath and make it fail if so.

### Details

As the group ID changes to "com.lightbend.akka" users may end up with two versions of this plugin on their classpath.
Find a way to detect that and fail.

**Comment by ennru** [comment]
One way to implement this is to rename the database provider class eg.
```
database-provider-fqcn = "akka.persistence.jdbc.util.DefaultSlickDatabaseProviderV4"
```
and add a `Class.forname` check for the old database provider class.

**Comment by renatocaval** [comment]
I'm moving this one to milestone 3.6.0. If we merge #329 and cut a 3.6.0, we must be able to detect if the old artefact is on the classpath. 

That said, using `DefaultSlickDatabaseProviderV4`, may not be the best option because we will be in 3.6.0

**Comment by renatocaval** [comment]
I did some trials and got some unsatisfactory results. 

It turns out that since the plugin is loaded dynamically, it's possible that the previous version get loaded and the check is never run. 

To prevent it, we will need to change the config namespace. Today it is defined at the root: `jdbc-journal`, `jdbc-snapshot-store` and `jdbc-read-journal`. 

We can move it to `akka-persistence-jdbc.write`, `akka-persistence-jdbc.snapshot` and `akka-persistence-jdbc.read` (this follows the couchbase plugin style).

Then, we should also rename the main plugin classes. For instance, class `akka.persistence.jdbc.journal.JdbcAsyncWriteJournal` must be renamed.

Here is what can still happen. 

1. we change the config namespace to `akka-persistence-jdbc.write` 
1. the user is loading the new config, so far so good
1. previous artefact is on the classpath. 
1. `akka-persistence-jdbc.write.class` points to `akka.persistence.jdbc.journal.JdbcAsyncWriteJournal`, but this class is also present on the previous artefact and classpath ordering make it load the wrong version

The result is that we end up loading the previous artefact and no check is run.

**Comment by renatocaval** [comment]
This creates a problem for fix #329. We would like to have a 3.6.0 release so users are not forced to migrate (see https://github.com/akka/akka-persistence-jdbc/issues/318). 

I'm fine to have #329 on a 3.6.0 release that requires some config change (because of namespace change as explained above), but that means that we can't upgrade Lagom with 3.6.0. It's safe to upgrade, but that should be an explicit action from the user.

So we have a few options here:

1. change the config and release 3.6.0 with #329. Lagom users being hit by the `eventsByTag` bug can manually upgrade to 3.6.0
2. create a 3.5.x branch using the last commit before we changed the groupId, merge #329 into it and cut a release

\#2 means that we cutting 3.5.3 using groupId `com.github.dnvriend` (needs @WellingR and @dnvriend helping us here). This release can be added to Lagom 1.6.1 and Lagom users get the fix automatically.

I'm just dropping the ideas here to know if someone has any other alternatives. I don't see how we could check the artefact version without changing the config namespace.

I'm inclined to \#1. The reason is that it requires less work with the drawback that Lagom users need to update manually. I don't find it a big deal though. The reality is that Lagom users don't immediately move to new releases. There lots of people still using 1.4 or 1.5 and at the end of the day, the majority will be upgrading the plugin manually anyway.

**Comment by renatocaval** [comment]
After some discussion, we decided to not change the config namespace. This has some impact. 

1) Akka users using 3.5.x and willing to move to 3.6.x will need to remove 3.5.x artifact from the classpath. If they have both and 3.5.x get loaded first, they will use 3.5.x and the check won't be run. Lagom users that upgrade manually to 3.6.0 will have the same issue.

There is a potential risk here tough. User may never load 3.6.x during development and Murphy law will make sure that 3.6.x gets loaded first in production, the check is run and BOOM!

2) Next Lagom version using 3.6.x will have a plugin extension to detect if both versions are on the classpath. This will cover Lagom users that for some reason have added 3.5.x explicitly on their dependencies. The plugin extension will enforce the check during development time.

**Comment by renatocaval** [comment]
@WellingR / @dnvriend, to avoid the risk of (1) happening, we would like to create a branch for 3.5.x, merge #329 and add a reverse of #335 to it. 

That means that we would release one last version under `com.github.dnvriend`. Would that work for you guys? 

The advantage is that whatever artifact is loaded, it will be able to detect the previous or the next version and force the user to choose one or another.

**Comment by WellingR** [comment]
Sounds good. I am able to push the release to bintray when it is ready. Dennis should still trigger one last maven sync

**Comment by renatocaval** [comment]
Great @WellingR. Thanks for that.

**Comment by renatocaval** [comment]
Fixed by #335
